### PR TITLE
fix edit form with attachments

### PR DESCRIPTION
**Pull request by snopoke**
If a form has attachments e.g. images then when the form is edited the attachments don't get copied over to the new form.

buddy @orangejenny

### CODE REVIEW

**Review by benrudolph — APPROVED** [review]
nice catch. thought for a second it might be related to this ticket: http://manage.dimagi.com/default.asp?258719#1374321. but alas i don't think those forms were edited

File: corehq/form_processor/parsers/form.py [hunk 1]
```diff
@@ -214,16 +214,7 @@ def apply_deprecation(existing_xform, new_xform, interface=None):
     #  - Save the new instance to the previous document to preserve the ID
 
     interface = interface or FormProcessorInterface(existing_xform.domain)
-
-    if existing_xform.persistent_blobs:
-        for name, meta in existing_xform.persistent_blobs.items():
-            with existing_xform.fetch_attachment(name, stream=True) as content:
-                existing_xform.deferred_put_attachment(
```
Comment (millerdev) [inline corehq/form_processor/parsers/form.py]: where these calls failing on SQL forms in the past? If not, why didn't it work for SQL forms?

**Review by millerdev — COMMENTED** [review]
It's a shame that SQL forms don't implement the same interface as Couch forms for attachments.

**Comment by snopoke** [comment]
@millerdev this code was originally added because when a form ID changes (as happens here in the edit form workflow) the references to the blobs is lost because it's based on the form ID. This code never executed for SQL forms since `form.persistent_blobs` is hardcoded as `None` in their models. It also never executed for Couch forms because the code I linked to already puts all the attachments in the `deferred` buffer and so `existing_xform.persistent_blobs` returns an empty dict.

This PR is addressing a different issue altogether where an edited forms loses the references to any attachments that it had.

**Comment by millerdev** [comment]
@snopoke, thanks for the explanation. That makes sense.

> This code never executed for SQL forms since `form.persistent_blobs` is hardcoded as `None` in their models. It also never executed for Couch forms because the code I linked to already puts all the attachments in the `deferred` buffer and so `existing_xform.persistent_blobs` returns an empty dict.

ouch

**Pull request merged**